appengine/pond_storage/rank.py

```diff
@@ -93,10 +99,36 @@ def create_ranking_match():
   # 4. Store match request in datastore.
   match = MatchRequest(entry_keys=entry_keys)
   match_key = match.put()
-  match_key_urlsafe = match_key.urlsafe()
-  logging.info('Created match request. matchKey:%s', match_key_urlsafe)
+  logging.info('Created match request. matchKey: %s', match_key)
   # 5. Return match request information.
-  return {'duck_list': duck_list, 'match_key': match_key_urlsafe}
+  return {'duck_list': duck_list, 'match_key': match_key.urlsafe()}
+
+def move_down_dummy_entry(dummy_entry, retries=5):
+  """Tries to move dummy entry down and/or deletes it if it is at bottom."""
+  @ndb.transactional(xg=True)
+  def swap_with_lower(lower_entry_key):
+    lower_entry = lower_entry_key.get()
+    # First confirm that lower entry rank is still farther down in leaderboard.
+    if not lower_entry and lower_entry.ranking > dummy_entry.key.get().ranking:
```

Comment:
A little confused by the logic here. It seems like we want to return False if the lower ranking is better than the dummy ranking ( which would be lower_entry.ranking < dummy_entry ranking) Seems like it should be:
if not lower_entry or lower_entry.ranking < dummy_entry.key.get().ranking?

Reply:
One tweak then lgtm. I think we don't need the not in front of the lower_entry. So would be if lower_entry and lower_entry.ranking....